public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -0,0 +1,483 @@
+  const dataQuerySpec = {
+    kind: getDataQueryKind(query),
+    spec: query,
+  };
+  return dataQuerySpec;
+}
+
+function getVizPanelTransformations(vizPanel: VizPanel): TransformationKind[] {
+  let transformations: TransformationKind[] = [];
+  const dataProvider = vizPanel.state.$data;
+  if (dataProvider instanceof SceneDataTransformer) {
+    const transformationList = dataProvider.state.transformations;
+    if (transformationList.length === 0) {
+      return [];
+    }
+    transformationList.forEach((transformationItem) => {
+      const transformation = transformationItem as DataTransformerConfig;
+      const transformationSpec: DataTransformerConfig = {
+        id: transformation.id,
+        disabled: transformation.disabled,
+        filter: {
+          id: transformation.filter?.id ?? '',
+          options: transformation.filter?.options ?? {},
+        },
+        topic: transformation.topic,
+        options: transformation.options,
+      };
+
+      transformations.push({
+        kind: transformation.id,
+        spec: transformationSpec,
+      });
+    });
+  }
+  return transformations;
+}
+
+function getVizPanelQueryOptions(vizPanel: VizPanel): QueryOptionsSpec {
+  let queryOptions: QueryOptionsSpec = {};
+  const queryRunner = getQueryRunnerFor(vizPanel);
+
+  if (queryRunner) {
+    queryOptions.maxDataPoints = queryRunner.state.maxDataPoints;
+
+    if (queryRunner.state.cacheTimeout) {
+      queryOptions.cacheTimeout = queryRunner.state.cacheTimeout;
+    }
+
+    if (queryRunner.state.queryCachingTTL) {
+      queryOptions.queryCachingTTL = queryRunner.state.queryCachingTTL;
+    }
+    if (queryRunner.state.minInterval) {
+      queryOptions.interval = queryRunner.state.minInterval;
+    }
+  }
+
+  const panelTime = vizPanel.state.$timeRange;
+
+  if (panelTime instanceof PanelTimeRange) {
+    queryOptions.timeFrom = panelTime.state.timeFrom;
+    queryOptions.timeShift = panelTime.state.timeShift;
+  }
+  return queryOptions;
+}
+
+function createElements(panels: PanelKind[]): Record<string, PanelKind> {
+  return panels.reduce(
+    (acc, panel) => {
+      const key = panel.spec.uid;
+      acc[key] = panel;
+      return acc;
+    },
+    {} as Record<string, PanelKind>
+  );
+}
+
+function getVariables(oldDash: DashboardSceneState) {
+  const variablesSet = oldDash.$variables;
+
+  // variables is an array of all variables kind (union)
+  let variables: Array<
+    | QueryVariableKind
+    | TextVariableKind
+    | IntervalVariableKind
+    | DatasourceVariableKind
+    | CustomVariableKind
+    | ConstantVariableKind
+    | GroupByVariableKind
+    | AdhocVariableKind
+  > = [];
+
+  if (variablesSet instanceof SceneVariableSet) {
+    variables = sceneVariablesSetToSchemaV2Variables(variablesSet);
+  }
+
+  return variables;
+}
+
+// Function to know if the dashboard transformed is a valid DashboardV2
+function isDashboardSchemaV2(dash: any): dash is DashboardV2 {
```

Comment:
Think it's fine for now to have this, tho the validation will be done on the API level.